packages/cozy-stack-client/src/KonnectorCollection.js

```diff
@@ -17,8 +24,73 @@ class KonnectorCollection extends AppCollection {
     throw new Error('destroy() method is not available for konnectors')
   }
 
-  async update() {
-    throw new Error('update() method is not available for konnectors')
+  async findTriggers(slug) {
+    const triggerCol = new TriggerCollection(this.stackClient)
+    const { data: rawTriggers } = await triggerCol.all()
+    return rawTriggers
+      .map(x => x.attributes)
+      .filter(triggerAttrs => isForKonnector(triggerAttrs, slug))
+  }
+
+  /**
+   * Launch a trigger for a given konnector.
+   *
+   * @param  {String} slug
+   * @param  {Object} options
+   * @param  {Object} options.accountId - Pinpoint the account that should be used, useful if the user
+   * has more than 1 account for 1 konnector
+   */
+  async launch(slug, options = {}) {
+    const triggerCol = new TriggerCollection(this.stackClient)
+    const konnTriggers = await this.findTriggers(slug)
+    const filteredTriggers = options.accountId
+      ? konnTriggers.filter(triggerAttrs =>
+          isForAccount(triggerAttrs, options.accountId)
+        )
+      : konnTriggers
+
+    if (filteredTriggers.length === 1) {
+      return triggerCol.launch(konnTriggers[0])
+    } else {
+      const filterAttrs = JSON.stringify(
+        pick({ slug, accountId: options.accountId })
+      )
+      if (filteredTriggers.length === 0) {
+        throw new Error(`No trigger found for ${filterAttrs}`)
+      } else if (filteredTriggers.length > 1) {
+        throw new Error(`More than 1 trigger found for ${filterAttrs}`)
+      }
+    }
+  }
+
+  /**
+   * Updates a konnector
+   *
+   * @param  {String} slug
+   * @param  {Object} options
+   * @param  {Object} options.source - Specify the source (ex: registry://slug/stable)
+   * @param  {Object} options.sync - Wait for konnector to be updated, otherwise the job
```

Comment:
I don't find `sync` is very clear. Is it a boolean?

Reply:
Yes, I fixed the type of the param, do you have any idea to make it clearer ?